Description: adjust initscript template
 Some tweaks to the initscript template are needed for Debian packaging to work
 and to follow policy.
Author: Philipp Huebner <debalance@debian.org>

--- a/ejabberd.init.template
+++ b/ejabberd.init.template
@@ -16,34 +16,52 @@
 
 set -o errexit
 
-DIR=@ctlscriptpath@
+. /lib/lsb/init-functions
+
+DIR=/usr/sbin
 CTL="$DIR"/ejabberdctl
-USER=@installuser@
+USER=ejabberd
+EJABBERDRUN=/run/ejabberd
 
 test -x "$CTL" || {
 	echo "ERROR: ejabberd not found: $DIR"
 	exit 1
 }
-getent passwd "$USER" >/dev/null || {
-	echo "ERROR: System user not found: $USER"
-	exit 2
+
+mkrundir()
+{
+   if [ ! -d $EJABBERDRUN ]; then
+       mkdir -p $EJABBERDRUN
+       if [ $? -ne 0 ]; then
+           echo -n " failed"
+           return
+       fi
+       chmod 0755 $EJABBERDRUN
+       chown ejabberd:ejabberd $EJABBERDRUN
+   fi
 }
 
+
 export PATH="${PATH:+$PATH:}/usr/sbin:/sbin"
 
 case "$1" in
   start)
     test -x "$CTL" || exit 0
+    mkrundir
     echo "Starting ejabberd..."
-    su - $USER -c "$CTL start"
-    su - $USER -c "$CTL started"
+    if ! /etc/init.d/ejabberd status > /dev/null; then
+      su - $USER -c "$CTL start"
+      su - $USER -c "$CTL started"
+    fi
     echo "done."
     ;;
   stop)
     test -x "$CTL" || exit 0
     echo "Stopping ejabberd..."
-    su - $USER -c "$CTL stop"
-    su - $USER -c "$CTL stopped"
+    if $CTL status >/dev/null || test $? = 1 ; then
+      su - $USER -c "$CTL stop"
+      su - $USER -c "$CTL stopped"
+    fi
     echo "done."
     ;;
   status)
